src/Arcus.Observability.Telemetry.Core/Logging/DependencyLogEntry.cs

```diff
@@ -0,0 +1,110 @@
+            int? resultCode,
+            bool isSuccessful,
+            IDictionary<string, object> context)
+        {
+            Guard.NotNullOrWhitespace(dependencyType, nameof(dependencyType), "Requires a non-blank custom dependency type when tracking the custom dependency");
+            Guard.NotLessThan(duration, TimeSpan.Zero, nameof(duration), "Requires a positive time duration of the dependency operation");
+            
+            DependencyType = dependencyType;
+            DependencyName = dependencyName;
+            DependencyData = dependencyData;
+            TargetName = targetName;
+            ResultCode = resultCode;
+            IsSuccessful = isSuccessful;
+
+            StartTime = startTime.ToString(FormatSpecifiers.InvariantTimestampFormat);
+            Duration = duration;
+            Context = context;
+        }
+
+        /// <summary>
+        /// Gets the custom type of the dependency.
+        /// </summary>
+        public string DependencyType { get; }
+        
+        /// <summary>
+        /// Gets the name of the dependency.
+        /// </summary>
+        public string DependencyName { get; }
+        
+        /// <summary>
+        /// Gets the custom data of the dependency.
+        /// </summary>
+        public object DependencyData { get; }
+        
+        /// <summary>
+        /// Gets the name of the dependency target.
+        /// </summary>
+        public string TargetName { get; }
+        
+        /// <summary>
+        /// Gets the code of the result of the interaction with the dependency.
+        /// </summary>
+        public int? ResultCode { get; }
+        
+        /// <summary>
+        /// Gets the indication whether or not the operation was successful.
+        /// </summary>
+        public bool IsSuccessful { get; }
+        
+        /// <summary>
+        /// Gets the point in time when the interaction with the HTTP dependency was started.
+        /// </summary>
+        public string StartTime { get; }
+        
+        /// <summary>
+        /// Gets the duration of the operation.
+        /// </summary>
+        public TimeSpan Duration { get; }
+        
+        /// <summary>
+        /// Gets the context that provides more insights on the dependency that was measured.
+        /// </summary>
+        public IDictionary<string, object> Context { get; }
+
+        /// <summary>
+        /// Returns a string that represents the current object.
+        /// </summary>
+        /// <returns>A string that represents the current object.</returns>
+        public override string ToString()
+        {
+            string contextFormatted = $"{{{String.Join("; ", Context.Select(item => $"[{item.Key}, {item.Value}]"))}}}";
+            return $"{DependencyType} {DependencyName} {DependencyData} named {TargetName} in {Duration} at {StartTime} " +
+                   $"(Successful: {IsSuccessful} - ResultCode: {ResultCode} - Context: {contextFormatted})";
+        }
```

Comment:
What are we doing with these 'ToStrings' ?

Reply:
Wel... the thing is... We can choose to use this Serilog formatting language, but that doesn't mean that every place it's used will be within Serilog. (That's not something we depent on or should be required when using our telemetry core package). So, if we don't wat our messages be: 
```csharp
Dependency Arcus.Observability.Telemetry.Core.Logging.DependencyLogTraget
```
we should have some way to represent it still as our 'original' sentence-like format, I guess.

Reply:
We are using our own type, but is the outcome the same if we:
1. Don't use a custom type
2. Don't override tostring

Because ideally we just allow people to use the standard approach imo

Reply:
It's not entirely the same as the log properties will probably not be available as separate parts, but the rendered message will be the same.

Reply:
Ok, no problem